Eastern-Research-Group/mywaterway
# Feature/165 improve monitoring performance

## Related Issues:
* https://jira.epa.gov/browse/HMW-165

## Main Changes:
* Updated the monitoring lists to use the VirtualizedList component. This improves performance of the tab after everything is loaded (i.e., scrolling through the list and clicking around).
* Updated the monitoringLocationsLayer to be a FeatureLayer instead of a GraphicsLayer. This improves the performance of loading or switching to a tab with monitoring data (i.e., Overview and Monitoring). 
  * The main performance issue with using a GraphicsLayer is with setting the popupTemplate. With the FeatureLayer, the popupTemplate is defined at the layer level whereas the popupTemplate is defined for each Graphic on a GraphicsLayer.
  * This change will also help us support point clustering for [HMW-160](https://jira.epa.gov/browse/HMW-160).
* Refactored the Sample Locations tab of the Monitoring panel. This was done to make the code simpler, more readable, and more performant. This code also still had remnants of the old class component code.

## Steps To Test:
1. Navigate http://localhost:3000/community/portland%20or/overview
2. Turn on the monitoring layer
3. Verify that performance (loading, scrolling through the list, expanding items, clicking stations on map, etc.) has been improved when compared to production
4. Click on the Monitoring panel
5. Verify that performance of switching to this tab has been improved, should be much faster now. Before it took about 40 - 60 seconds and now it takes about 2 seconds
6. Repeat steps 2 and 3
7. Check performance of togging filtering switches on/off, this should be much faster now. Before it could take up to 30 seconds and now it takes about 2 seconds.
8. Search for a different location
9. Verify that the monitoring stations for portland oregon are no longer on the map

## TODO:
- [x] PR #601 needs to be merged in before this one can be merged.
